slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -216,6 +226,18 @@ ::grpc::Status SLAMServiceImpl::GetJpegMap(const GetMapRequest *request,
 
 ::grpc::Status SLAMServiceImpl::GetCurrentPointCloudMap(
     const GetMapRequest *request, GetMapResponse *response) {
+    // if we have a map for localization, use it
+    if (localization_map_ready) {
```

Comment:
I did not turn this into a helper function because the `GetMap` endpoint should be removed soon, but I can go ahead and make a function based off your other feedback

Reply:
I remembered the real reason, `GetMap` and `GetPointCloudMap` have two different response types, so I would need to make a different function for each one.